seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java

```diff
@@ -0,0 +1,309 @@
+    }
+
+    public void createDir(@NonNull String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        Path dfs = new Path(filePath);
+        if (!fileSystem.mkdirs(dfs)) {
+            throw new FileConnectorException(
+                    CommonErrorCodeDeprecated.FILE_OPERATION_FAILED,
+                    "create dir " + filePath + " error");
+        }
+    }
+
+    public List<Path> getAllSubFiles(@NonNull String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        List<Path> pathList = new ArrayList<>();
+        if (!fileExist(filePath)) {
+            return pathList;
+        }
+        Path fileName = new Path(filePath);
+        FileStatus[] status = fileSystem.listStatus(fileName);
+        if (status != null) {
+            for (FileStatus fileStatus : status) {
+                if (fileStatus.isDirectory()) {
+                    pathList.add(fileStatus.getPath());
+                }
+            }
+        }
+        return pathList;
+    }
+
+    public FileStatus[] listStatus(String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        return fileSystem.listStatus(new Path(filePath));
+    }
+
+    public FileStatus getFileStatus(String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        return fileSystem.getFileStatus(new Path(filePath));
+    }
+
+    public FSDataOutputStream getOutputStream(String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        return fileSystem.create(new Path(filePath), true);
+    }
+
+    public FSDataInputStream getInputStream(String filePath) throws IOException {
+        if (fileSystem == null) {
+            initialize();
+        }
+        return fileSystem.open(new Path(filePath));
+    }
+
+    @SneakyThrows
+    public <T> T doWithHadoopAuth(HadoopLoginFactory.LoginFunction<T> loginFunction) {
+        if (configuration == null) {
+            this.configuration = createConfiguration();
+        }
+        if (enableKerberos()) {
+            configuration.set("hadoop.security.authentication", "kerberos");
+            return HadoopLoginFactory.loginWithKerberos(
+                    configuration,
+                    hadoopConf.getKrb5Path(),
+                    hadoopConf.getKerberosPrincipal(),
+                    hadoopConf.getKerberosKeytabPath(),
+                    loginFunction);
+        }
+        if (enableRemoteUser()) {
+            return HadoopLoginFactory.loginWithRemoteUser(
+                    configuration, hadoopConf.getRemoteUser(), loginFunction);
+        }
+        return loginFunction.run(configuration, UserGroupInformation.getCurrentUser());
+    }
+
+    @Override
+    public void close() throws IOException {
+        synchronized (UserGroupInformation.class) {
```

Comment:
I think there is a lock based on UserGroupInformation within the logoutUserFromKeytab method. Do we need to expand the scope of the lock?

Reply:
Good catch, this lock is not needed, I removed this lock.